src/charm.py

```diff
@@ -246,25 +251,33 @@ def _reconcile(self) -> None:
                 tls_information = TLSInformation.from_charm(self, self.certificates)
                 self._tls.certificate_available(tls_information)
 
-                # Get peer units information
-                peers: list[str] = []
-                unit_address = self._get_unit_address()
-                if not unit_address:
-                    self.unit.status = ops.BlockedStatus(
-                        "Couldn't get the executing unit's IP address."
+                haproxy_route_requirers_information = (
+                    HaproxyRouteRequirersInformation.from_provider(
+                        self.haproxy_route_provider,
+                        tls_information,
+                        self._get_peer_units_address(),
                     )
-                    return
-                peers.append(unit_address)
-                peers.extend(self._get_peer_units_address())
-
-                haproxy_route_requirers_information = HaproxyRouteRequirersInformation.from_charm(
-                    self.haproxy_route_provider, tls_information, peers
                 )
                 self.haproxy_service.reconcile_haproxy_route(
                     config, haproxy_route_requirers_information
                 )
                 self.unit.set_ports(80, 443)
+                if self.unit.is_leader():
+                    for backend in haproxy_route_requirers_information.backends:
+                        relation = self.model.get_relation(
+                            HAPROXY_ROUTE_RELATION, backend.relation_id
+                        )
+                        if not relation:
+                            logger.error("Relation does not exist, skipping.")
+                        self.haproxy_route_provider.publish_proxied_endpoints(
+                            [f"https://{hostname}" for hostname in backend.hostname_acls], relation
```

Comment:
We should provide the full urls including the path ACLs

Reply:
updated to send the full urls